### DISCUSSION TITLE
vscode-premake-tools

### DISCUSSION THREAD

**Discussion by AbdullahAmrSobh** [premake/premake-core#1695] [General]
I was thinking it would be nice if premake had a vscode integration, similar to vscode-cmake-tools.
that can detect installed compilers, configure language servers, setup build tasks and etc.

**Comment by Jarod42** [comment]
I saw
- https://github.com/shoxsz/proj-maker-cpp-vscode-premake
- https://github.com/Cannedfood/premake-vscode

Not sure how there are usable (or not).

**Reply by Cannedfood** [reply]
shoxsz's projects seems to be mostly for generating premake+vscode projects. (Not an extension, more of a premake5.lua and .vscode/tasks.json generator).

Mine is definetely not usable yet, I haven't worked on it for a while. Right now it basically just adds some snippets.
